### PR TITLE
logout feature added

### PR DESCRIPTION
**Pull request by puneeth2001**
Now a user can login , use the portal and then logout

### CODE REVIEW

**Comment by puneeth2001** [comment]
@aswinshenoy @vchrombie can you please review this.

**Comment by aswinshenoy** [comment]
@puneeth2001 Please send multiple pull requests for multiple features. As a reviewer, it becomes really hard to review and verify changes when a lot of things are changed together.

**Comment by puneeth2001** [comment]
@aswinshenoy sorry, I will make sure that this will not happen from the next PRs

File: portal/settings.py [hunk 1]
```diff
@@ -25,7 +25,7 @@
 # SECURITY WARNING: don't run with debug turned on in production!
 DEBUG = True
 
-ALLOWED_HOSTS = []
+ALLOWED_HOSTS = ['127.0.0.1']
```
Comment (aswinshenoy) [inline portal/settings.py]: It's not time yet, to decide what all should be in allowed_hosts. Infact, I hope you understand why that setting exists. Perhaps, we will be creating `server_settings.py` and `local_settings.py` just like cms once we are ready for deployment.

File: portal/urls.py [hunk 2]
```diff
@@ -19,5 +19,6 @@
 urlpatterns = [
     path('admin/', admin.site.urls),
     path('',include('user.urls')),
-    path('repo/', include('repo.urls'))
+    path('repo/', include('repo.urls')),
+    path('accounts/', include('django.contrib.auth.urls')),
```
Comment (aswinshenoy) [inline portal/urls.py]: can you provide a screenshot of what `accounts/` looks like, and why you needed that?

File: portal/settings.py [hunk 3]
```diff
@@ -102,7 +105,15 @@
     },
 ]
 
+AUTHENTICATION_BACKENDS = (
+   # 'social_core.backends.open_id.OpenIdAuth',  # for Google authentication
+   # 'social_core.backends.google.GoogleOpenId',  # for Google authentication
+   # 'social_core.backends.google.GoogleOAuth2',  # for Google authentication
+    'social_core.backends.github.GithubOAuth2',  # for Github authentication
+    #'social_core.backends.facebook.FacebookOAuth2',  # for Facebook authentication
```
Comment (aswinshenoy) [inline portal/settings.py]: Please remove everything except Github, as there is never a case anything else would be required.

**Comment by aswinshenoy** [comment]
> @aswinshenoy sorry, I will make sure that this will not happen from the next PRs

Also make sure you have a clear name for the PR, and you dont keep changing it as well.

This PR essential does 2 things
1. Removes adding that extra portal folder (which should have never been merged, and actually fixed through rebasing/reverting)
2.  Implements Github Login

Make sure your title is clear for everyone from next time. :)

File: portal/urls.py (hunk 2 above)
Comment (puneeth2001) [reply portal/urls.py]: I used it for login feature:
screenshot-127 0 0 1-8900-2019 07 02-13-59-50

File: portal/settings.py (hunk 1 above)
Comment (puneeth2001) [reply portal/settings.py]: okay

File: portal/urls.py (hunk 2 above)
Comment (aswinshenoy) [reply portal/urls.py]: where is Github Login ?

Comment (aswinshenoy) [reply portal/urls.py]: and why is is called `accounts/`? It should be `/login` right?

File: portal/settings.py (hunk 3 above)
Comment (puneeth2001) [reply portal/settings.py]: done

**Pull request merged**